keps/sig-storage/20190122-csi-inline-volumes.md

```diff
@@ -209,44 +140,38 @@ type CSIVolumeSource struct {
-	// sensitive information to pass to the CSI driver to complete the CSI
-	// ControllerPublishVolume and ControllerUnpublishVolume calls.
-	// This field is optional, and  may be empty if no secret is required. If the
-	// secret object contains more than one secret, all secret references are passed.
-	// 
-	// +optional
-	ControllerPublishSecretRef *LocalObjectReference
-
-	// NodeStageSecretRef is a reference to the secret object containing sensitive
-	// information to pass to the CSI driver to complete the CSI NodeStageVolume
-	// and NodeStageVolume and NodeUnstageVolume calls.
-	// This field is optional, and  may be empty if no secret is required. If the
-	// secret object contains more than one secret, all secret references are passed.
-	// 
-	// +optional
-	NodeStageSecretRef *LocalObjectReference
-
 	// NodePublishSecretRef is a reference to the secret object containing
 	// sensitive information to pass to the CSI driver to complete the CSI
 	// NodePublishVolume and NodeUnpublishVolume calls.
 	// This field is optional, and  may be empty if no secret is required. If the
 	// secret object contains more than one secret, all secret references are passed.
-	// 
 	// +optional
 	NodePublishSecretRef *LocalObjectReference
 }
 ```
 
-### Secret references
-Secret references declared in an inline CSI volume can only be used with namespaces from pods where they are referenced .  For inline usage, secret references are stored in `LocalObjectReference` values:
-* `LocalObjectReference` do not include a namespace reference.  This is to prevent reference to arbitrary namespace values.
-* The namespace reference will be extracted from the the pod spec at different phases of storage lifecycle by the Kubelet or external CSI compoennts.
-* The Kubelet and external CSI components must ensure secret references can only be used with the namespace from inline pod spec.
+### Driver mode
+To indicate that the driver will support ephemeral inline volume requests, the existing `CSIDriver` object will be extended to include attribute `Mode`.  Currently the only modes that will be supported are `persistent` and `ephemeral`.  
 
+When `CSIDriver.Mode == <not specified>` or when `CSIDriver.Mode == persistent`, the driver will function as normal supporting only PV/PVC-requested volumes and
+will receive all persistent volume operation calls (i.e. provision/delete, attach/detach, mount/unmount, etc).
+
+When `CSIDriver.Mode == ephemeral` the followings are assumed:
+* Volume requests will originate from pod specs.
+* The driver will only receive volume operation calls during mount/unmount phase.
+* The driver will not receive separate gRPC calls for provisioning, attaching, detaching, and deleting of volumes.
+* The driver is responsible for implementing steps to ensure the volume is created and made avaiable to the pod during mount call
+* The driver is responsible for implementing steps to delete and clean up any volume and resources during the unmount call.
+
```

Comment:
Maybe two more while we're at it:
* The mount could happen multiple times for the same volumeHandle. The first should allocate all backing volume data.
* The unmount could happen multiple times for the same volumeHandle. The first should remove any backing volume data. It will be called at a minimum one time. No volume accesses will be made after the first call.

Reply:
@kfox1111 I will highlight your points.